readthedocs/projects/tasks/search.py

```diff
@@ -120,7 +121,36 @@ def collect(self, sync_id: int):
         self.version.imported_files.exclude(build=sync_id).delete()
 
 
-def _get_indexers(*, version, search_ranking, search_ignore, search_index_name=None):
+class FileManifestIndexer(Indexer):
+    def __init__(self, version: Version, build: Build):
+        self.version = version
+        self.build = build
+        self._hashes = {}
+
+    def process(self, html_file: HTMLFile, sync_id: int):
+        self._hashes[html_file.path] = html_file.processed_json["main_content_hash"]
```

Comment:
I do wonder if we want to be storing the hash on the HTMLFile as well. With this architecture, querying the hash of a file is going to be pretty complex and require pulling from S3. I'm not sure if we have an obvious use case for getting the hash of a specific file, so maybe something to note for the future?

```suggestion
        self._hashes[html_file.path] = html_file.processed_json["main_content_hash"]
        # TODO: Possibly store the hash on the html_file object,
        # since this will allow us to query that from the DB directly.
```

Reply:
> Alternatively, we could store it in redis/cache, since the query is likely to happen pretty good after creation 🤔

Good point. Most of projects using PR builder will access the PR at least once immediately after the build finishes. We could consider optimizing this somehow (eg. saving the whole JSON in the cache for 30m --so, we don't have to hit S3); but I'm not too worried about this just yet. Dealing with cache is always hard due to invalidation 😄 

It's a good idea to leave a comment here with these ideas, tho 👍🏼